### PR TITLE
Add a 'name' argument to '.copy' method of Theano variables

### PR DESCRIPTION
**Pull request by yingzha**

### CODE REVIEW

**Comment by yingzha** [comment]
@dwf I modified the arg in pprint.assign, I guess it will pass the Travis.

File: theano/tensor/basic.py [hunk 1]
```diff
@@ -2815,9 +2815,11 @@ def do_constant_folding(self, node):
 pprint.assign(alloc, printing.FunctionPrinter('alloc'))
 
 
-"""Create a duplicate of `a` (with duplicated storage)"""
-tensor_copy = elemwise.Elemwise(scal.identity)
-pprint.assign(tensor_copy, printing.IgnorePrinter())
+def tensor_copy(input):
```
Comment (dwf) [inline theano/tensor/basic.py]: I don't think this change is necessary?

File: theano/tensor/var.py [hunk 2]
```diff
@@ -520,8 +520,11 @@ def take(self, indices, axis=None, mode='raise'):
         return theano.tensor.subtensor.take(self, indices, axis, mode)
 
     # COPYING
-    def copy(self):
-        return theano.tensor.basic.tensor_copy(self)
+    def copy(self, name=None):
+        """Copy a variable and set a name to the copy."""
```
Comment (dwf) [inline theano/tensor/var.py]: "and optionally assign a name"

**Comment by dwf** [comment]
This should probably be rebased and I think one of the changes is irrelevant (the change to `tensor_copy`). Otherwise it is fine.

File: theano/tensor/basic.py (hunk 1 above)
Comment (yingzha) [inline theano/tensor/basic.py]: OK, I'll rebate it

**Comment by nouiz** [comment]
You don't need to create a new PR when you rebase. You can force the push
again your github branch.

**Comment by yingzha** [comment]
@nouiz Since I made changes on master branch before, I'm not sure if I can rebate it directly. That's why I copied the changes to another branch and closed this one.

**Pull request closed without merging**